enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -381,4 +378,29 @@ private ClassPathResult findBomClasspath(
     }
     return result;
   }
+
+  /**
+   * Returns the class path entries that contains entry point classes.
+   *
+   * <p>For BOM projects, they are the dependencies listed in the {@code dependencyManagement}
+   * section.
+   *
+   * <p>For normal projects, they are the direct dependencies listed in the {@code dependencies}
+   * section.
+   */
+  private ImmutableList<ClassPathEntry> entryPoints(
+      MavenProject project, ImmutableList<ClassPathEntry> classPath) {
+    // As sorted by level order, the first elements in classpath are the project and its direct
+    // non-test dependencies.
+    List<org.apache.maven.model.Dependency> directDependencies =
+        dependencySection == DependencySection.DEPENDENCY_MANAGEMENT
+            ? project.getDependencyManagement().getDependencies()
+            : project.getDependencies();
```

Comment:
This is the fix. The entry point JARs are the direct dependencies for non-BOM project, and the ones in dependencyManagement sections for BOM project.